### ISSUE TITLE
Hangs after minimizing window on Windows

### ISSUE THREAD

**Issue by Jascha-N** [PistonDevelopers/conrod#892]
The `text`, `text_edit` and all three variants of the `all_widgets` examples hang after minimizing the window. The CPU usage shoots up to maximum and restoring the window will leave it blank and unresponsive. In my own program I've been able to pinpoint it down to updating a `TitleBar` on a dynamically sized sub-`Canvas` (`flow_right`). It gets stuck on the `set` method. I think the problem might have something to do with text rendering.

**Comment by mitchmindtree** [comment]
@Jascha-N hey thanks for the issue.

Out of curiousity, does the glutin_glium.rs example have this issue?

Also, does this hang during minimize occur for any of the upstream [winit](https://github.com/tomaka/winit) examples? Does it occur when running the [rusttype](https://github.com/dylanede/rusttype) gpu_cache.rs example?

Sorry for all the Qs, just need to gauge whether the problem might be upstream or not first.

**Comment by Jascha-N** [comment]
`all_widgets`, `glutin_glium` and `glutin_gfx` (also renders incorrectly) all have the same issue. My application uses `glutin`+`glium` as the backend for conrod and has the same problem.

I tried running all of the examples and there were no issues, outside of the `rusttype` example crashing in release mode because of an invalid pointer passed to `CreateWindowEx` (a pre-`winit` `glutin` issue I encountered before).

Commenting out the `.set` for that one particular `TitleBar` (other `TitleBar`s work fine) stops the main thread from hanging.

**Comment by Jascha-N** [comment]
I have logged the `glutin` events and this is what I see when minimizing:

```
[INFO][communicator::ui] Event received: Focused(false)
[INFO][communicator::ui] Event received: Moved(33536, 33536)
[INFO][communicator::ui] Event received: Resized(0, 0)
```

The window gets resized to 0x0. Could this be the culprit?

**Comment by Jascha-N** [comment]
As a side note, and correct me if I'm wrong, but I think the `glutin` back-end still uses an old version (0.6.x) which doesn't even depend on `winit` yet.

**Comment by mitchmindtree** [comment]
> As a side note, and correct me if I'm wrong, but I think the glutin back-end still uses an old version (0.6.x) which doesn't even depend on winit yet.

Yeah this is true, we're currently blocked on updating to a newer version as the latest version of glium still depends on glutin 0.6.x.

**Comment by Jascha-N** [comment]
I see, glium hasn't been updated on crates.io in a while :(
